source/guide_codimd.rst

```diff
@@ -0,0 +1,303 @@
+    "production": {}
+  }
+
+That’s all. The actual configuration will be done through environment!
+
+Configure port
+--------------
+
+Since CodiMD uses its own webserver, you need to find a free port and bind your application to it.
+
+.. include:: includes/generate-port.rst
+
+
+Get Session Cookie
+------------------
+
+Generate random string to sign our session cookies with:
+
+.. code-block:: console
+
+  [isabell@stardust ~]$ < /dev/urandom tr -dc _A-Z-a-z-0-9 | head -c${1:-32};echo;
+  extremerandom
+  [isabell@stardust ~]$
+
+And remember it.
+
+Setup daemon
+------------
+
+Create a file ``~/etc/services.d/codemd.ini`` and put the following in it:
+
+.. code-block:: ini
+
+	[supervisord]
+	loglevel=warn
+
+	[program:codimd]
+	environment=
+		CMD_SESSION_SECRET="<random>",
+		CMD_DOMAIN="<domain>",
+		CMD_PORT=<port>,
+		NODE_ENV="production",
+		CMD_USECDN=false,
+		CMD_ALLOW_ANONYMOUS=false,
+		CMD_ALLOW_ANONYMOUS_EDITS=true,
+		CMD_ALLOW_FREEURL=true,
+		CMD_DEFAULT_PERMISSION=private,
+		CMD_SESSION_LIFE=1209600000,
+		CMD_EMAIL=true,
+		CMD_ALLOW_EMAIL_REGISTER=false,
+		CMD_IMAGE_UPLOAD_TYPE=filesystem,
+		CMD_PROTOCOL_USESSL=true,
+		CMD_DB_URL="mysql://<username>:<mysql_pw>@localhost:3306/<username>_codimd",
+
+	directory=/home/<username>/codimd
+	command=node app.js
+
+In our example this would be:
+
+.. code-block:: ini
+
+	[supervisord]
+	loglevel=warn
+
+	[program:codimd]
+	environment=
+		CMD_SESSION_SECRET="extremerandom",
+		CMD_DOMAIN="isabell.uber.space",
+		CMD_PORT=9000,
+		NODE_ENV="production",
+		CMD_USECDN=false,
+		CMD_ALLOW_ANONYMOUS=false,
+		CMD_ALLOW_ANONYMOUS_EDITS=true,
+		CMD_ALLOW_FREEURL=true,
+		CMD_DEFAULT_PERMISSION=private,
+		CMD_SESSION_LIFE=1209600000,
+		CMD_EMAIL=true,
+		CMD_ALLOW_EMAIL_REGISTER=false,
+		CMD_IMAGE_UPLOAD_TYPE=filesystem,
+		CMD_PROTOCOL_USESSL=true,
+		CMD_DB_URL="mysql://isabell:MySuperSecretPassword@localhost:3306/isabell_codimd",
+
+	directory=/home/isabell/codimd
+	command=node app.js
+
+
+
+Replace the values in ``CMD_SESSION_SECRET``, ``CMD_DOMAIN``, ``CMD_PORT`` and ``CMD_DB_URL`` and you're good to go!
+
+See `here <https://github.com/hackmdio/codimd#environment-variables-will-overwrite-other-server-configs>`_ for a detailed look at what these options do.
+
+
+Once you run the following step, this will create a CodiMD instance without reliance on third-party CDN servers, which is private to people who had an account created via CLI.
+
+
+
+Finishing installation
+======================
+
+.. code-block:: console
+
+  [isabell@stardust ~]$ supervisorctl reread
+  [isabell@stardust ~]$ supervisorctl update
+  [isabell@stardust ~]$
+
+Add a user
+----------
+
+Since we don’t want to run a public instance and have disabled registration, we use the CLI to add a user:
+
+.. code-block:: console
+
+  [isabell@stardust ~]$ cd ~/codemd
+  [isabell@stardust ~]$ CMD_DB_URL="mysql://<username>:<mysql_pw>@localhost:3306/<username>_codimd" bin/manage_users --add isabell@uberspace.de
```

Comment:
working directory should now be ~/codimd, so the line should start with `[isabell@stardust codimd]$`

Reply:
Oh, and the email address should be isabell@uber.space

Reply:
Unfortunately, the command produces only an error for me, not prompting for a password.

```
[testnico@stardust codimd]$ CMD_DB_URL="mysql://testnico:CENSORED@localhost:3306/testnico_codimd" bin/manage_users --add nico@team.uber.space
2018-10-12T16:56:50.942Z - error: uncaughtException: Cannot find module './build/Release/scrypt' date=Fri Oct 12 2018 16:56:50 GMT+0000 (UTC), pid=27308, uid=1015, gid=1016, cwd=/home/testnico/codimd, execPath=/opt/nodejs8/bin/node, version=v8.11.3, argv=[/opt/nodejs8/bin/node, /home/testnico/codimd/bin/manage_users, --add, nico@team.uber.space], rss=51535872, heapTotal=39235584, heapUsed=24581416, external=405995, loadavg=[0.0029296875, 0.15576171875, 0.52978515625], uptime=564884, trace=[column=15, file=module.js, function=Module._resolveFilename, line=547, method=_resolveFilename, native=false, column=25, file=module.js, function=Module._load, line=474, method=_load, native=false, column=17, file=module.js, function=Module.require, line=596, method=require, native=false, column=18, file=internal/module.js, function=require, line=11, method=null, native=false, column=20, file=/home/testnico/codimd/node_modules/scrypt/index.js, function=null, line=3, method=null, native=false, column=30, file=module.js, function=Module._compile, line=652, method=_compile, native=false, column=10, file=module.js, function=Module._extensions..js, line=663, method=.js, native=false, column=32, file=module.js, function=Module.load, line=565, method=load, native=false, column=12, file=module.js, function=tryModuleLoad, line=505, method=null, native=false, column=3, file=module.js, function=Module._load, line=497, method=_load, native=false, column=17, file=module.js, function=Module.require, line=596, method=require, native=false, column=18, file=internal/module.js, function=require, line=11, method=null, native=false, column=14, file=/home/testnico/codimd/lib/models/user.js, function=null, line=4, method=null, native=false, column=30, file=module.js, function=Module._compile, line=652, method=_compile, native=false, column=10, file=module.js, function=Module._extensions..js, line=663, method=.js, native=false, column=32, file=module.js, function=Module.load, line=565, method=load, native=false], stack=[Error: Cannot find module './build/Release/scrypt',     at Function.Module._resolveFilename (module.js:547:15),     at Function.Module._load (module.js:474:25),     at Module.require (module.js:596:17),     at require (internal/module.js:11:18),     at Object.<anonymous> (/home/testnico/codimd/node_modules/scrypt/index.js:3:20),     at Module._compile (module.js:652:30),     at Object.Module._extensions..js (module.js:663:10),     at Module.load (module.js:565:32),     at tryModuleLoad (module.js:505:12),     at Function.Module._load (module.js:497:3),     at Module.require (module.js:596:17),     at require (internal/module.js:11:18),     at Object.<anonymous> (/home/testnico/codimd/lib/models/user.js:4:14),     at Module._compile (module.js:652:30),     at Object.Module._extensions..js (module.js:663:10),     at Module.load (module.js:565:32)]
```